pkg/apis/v1beta1/nodepool_status.go

```diff
@@ -22,6 +22,9 @@ import (
 
 // NodePoolStatus defines the observed state of NodePool
 type NodePoolStatus struct {
+	// Nodes is the number of nodes maintained by ths NodePool
+	// +optional
+	Nodes int `json:"nodes,omitempty"`
```

Comment:
Why not just make this part of resources? I don't think you need to make an API change for this?